src/dos/drive_virtual.cpp

```diff
@@ -22,41 +22,158 @@
 #include <stdlib.h>
 #include <string.h>
 #include <time.h>
+#include <fstream>
 
 #include "dos_inc.h"
 #include "support.h"
+#include "shell.h"
 #include "cross.h"
 
+#define default_date DOS_PackDate(2002, 10, 1)
+#define default_time DOS_PackTime(12, 34, 56)
+
+struct Filename {
+	std::string fullname = {};
+	std::string shortname = {};
+};
+std::vector<Filename> vfilenames = {Filename{"", ""}};
+constexpr int max_vfiles = 500;
+unsigned int vfile_pos = 1;
+uint16_t fztime = 0;
+uint16_t fzdate = 0;
+char sfn[DOS_NAMELENGTH_ASCII];
+void Add_VFiles(const bool add_autoexec);
+extern DOS_Shell *first_shell;
+
 struct VFILE_Block {
-	const char * name;
-	Bit8u * data;
-	Bit32u size;
-	Bit16u date;
-	Bit16u time;
-	VFILE_Block * next;
+	const char * name = nullptr;
+	uint8_t * data = nullptr;
+	uint32_t size = 0;
+	uint16_t date = 0;
+	uint16_t time = 0;
+	unsigned int onpos = 0;
+	bool isdir = false;
+	VFILE_Block * next = nullptr;
 };
 
+static VFILE_Block *first_file = nullptr;
+static VFILE_Block *parent_dir = nullptr;
 
-static VFILE_Block * first_file;
+char *VFILE_Generate_8x3(const char *name, const unsigned int onpos)
+{
+	if (!name || !*name) {
+		strcpy(sfn, "");
+		return sfn;
```

Comment:
We have a couple of helper functions to safely terminate C-strings, so a very minor suggestion is to use `reset_str(sfn)` here instead.

![2022-03-18_12-10](https://user-images.githubusercontent.com/1557255/159069144-78256fbd-786f-4861-b1a3-db6e89a61fc2.png)

Reply:
Changed to reset_str(sfn) as suggested.